docs(alert-dialog): updated docs (#5371)
adobe/spectrum-web-components: packages/alert-dialog/README.md

## Overview

`<sp-alert-dialog>` displays important information that users need to acknowledge. When used directly, the `<sp-alert-dialog>` element surfaces a `slot` based API for deep customization of the content to be included in the overlay.

### Usage

[![See it on NPM!](https://img.shields.io/npm/v/@spectrum-web-components/alert-dialog?style=for-the-badge)](https://www.npmjs.com/package/@spectrum-web-components/alert-dialog)
[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/alert-dialog?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/alert-dialog)

```bash
yarn add @spectrum-web-components/alert-dialog
```

Import the side effectful registration of `<sp-alert-dialog>` via:

```javascript
import '@spectrum-web-components/alert-dialog/sp-alert-dialog.js';
```

When looking to leverage the `AlertDialog` base class as a type and/or for extension purposes, do so via:

```javascript
import { AlertDialog } from '@spectrum-web-components/alert-dialog';
```

### Anatomy

The alert dialog consists of several key parts:

> -   A heading, using `slot="heading"`, that describes the purpose of the dialog

Reviewer: ```suggestion
-   All alert dialogs must have a title, using `slot="heading"`, that uses a few words to convey the outcome of what will happen if a user continues with an action
```

> -   Content, using the default slot, that provides additional context

Reviewer: ```suggestion
-   **Description:** Alert dialogs can include a description using the default slot. A description briefly communicates any additional information or context that a user needs to know to continue with an action
```

> -   Action buttons, using `slot="button"`, that allow users to respond

Reviewer: ```suggestion
-   **Button:**
     -   Use `slot="button"` to render your action button(s) that allow users to respond
     -   An alert dialog must have one primary action button with the option to include a secondary action and cancel action.
     - Non-primary action buttons should be `variant="secondary"` and `treatment: "outline"`.
     -   The three buttons should be rendered in the DOM in the following order:
          -   **Cancel action:** Offers an option to go back and cancel the action.
          -   **Secondary action:** Offers a secondary action. e.g. "Remind me later"
          -   **Primary action:**  The first (right-most) button communicates what the button will do if selected, or to acknowledge and dismiss the dialog. Check [variants](#variants) for the correct primary button styling.
```

Reviewer 2: Hey, I'm wondering if having 3 buttons in the dialog is the right approach. Would you mind checking in with Nate Baldwin about this? I believe he's working on a more comprehensive dialog RFC that addresses a lot of the related concerns. It might offer some useful guidance and help us align better with the direction he's setting.

```html
<sp-alert-dialog variant="confirmation">
    <h2 slot="heading">Important Notice</h2>
    <p>This action requires your confirmation.</p>
    <sp-button
        slot="button"
        variant="secondary"
        treatment="outline"
        onclick="this.dispatchEvent(new Event('close', { bubbles: true, composed: true }));"
    >
        Cancel
    </sp-button>
    <sp-button
        slot="button"
        variant="accent"
        treatment="fill"
        onclick="this.dispatchEvent(new Event('close', { bubbles: true, composed: true }));"
    >
        Confirm
    </sp-button>
</sp-alert-dialog>
```

### Options

#### Variants

> The alert dialog supports different variants to convey the nature and importance of the message:

Reviewer: ```suggestion
The alert dialog supports `confirmation`, `information`, `warning`, `error`, and `destructive` variants to convey the nature and importance of the message:
```

    </sp-button>
    <sp-button
        slot="button"
        variant="accent"
        treatment="fill"
        onclick="this.dispatchEvent(new Event('close', { bubbles: true, composed: true }));"
    >
        Confirm
    </sp-button>
</sp-alert-dialog>
```

#### Buttons

Use `slot="button"` to render your action button(s) that allow users to respond

> -   An alert dialog must have one primary action button (with `variant="secondary"`) with the option to include a secondary action and/or a cancel action.

Author: Nice catch, @cdransf! Pushing a correction.

> -   Non-primary action buttons should be `variant="secondary"` and `treatment: "outline"`.

Reviewer 3: Should this be `treatment="outline"`?

-   The three buttons should be rendered in the DOM in the following order:
    -   **Cancel action:** Offers an option to go back and cancel the action.
    -   **Secondary action:** Offers a secondary action. e.g. "Remind me later"

>     -   **Primary action:** The first (right-most) button communicates what the button will do if selected, or to acknowledge and dismiss the dialog. Check [variants](#variants) for the correct primary button styling. See also the (Alert Dialog design options)[https://spectrum.adobe.com/page/alert-dialog/#Options].

Reviewer 3: I think this link syntax should be swapped. 

```
[Alert Dialog design options](https://spectrum.adobe.com/page/alert-dialog/)
```

    >
        Delete
    </sp-button>
</sp-alert-dialog>
```

</sp-tab-panel>
</sp-tabs>

### Behaviors

#### Context

An alert dialog should be placed inside a model overaly:

```html
<sp-button id="trigger">open modal</sp-button>
<sp-overlay trigger="trigger@click" type="modal">
    <sp-alert-dialog variant="confirmation">
        <h2 slot="heading">Important Notice</h2>
        <p>This action requires your confirmation.</p>
        <sp-button
            slot="button"
            variant="secondary"
            treatment="outline"
            onclick="this.dispatchEvent(new Event('close', { bubbles: true, composed: true }));"
        >
            Cancel
        </sp-button>
        <sp-button
            slot="button"
            variant="accent"
            treatment="fill"
            onclick="this.dispatchEvent(new Event('close', { bubbles: true, composed: true }));"
        >
            Confirm
        </sp-button>
    </sp-alert-dialog>
</sp-overlay>
```

### Accessibility

When implementing an alert dialog:

-   Use concise, meaningful dialog title that clearly states the purpose

> -   Uses semantic heading elements (`<h2>`) for the dialog title

Reviewer 2: ```suggestion
-   Use semantic heading elements (`<h2>`) for the dialog title
```

-   Ensure button labels clearly indicate the action they will perform

> -   Provides clear, concise content that explains the situation and required actions

Reviewer 2: ```suggestion
-   Provide clear, concise content that explains the situation and required actions
```

> -   Consider the appropriate variant based on the message's importance and urgency

Reviewer 3: I love this section here. Should we add any suggestions on using `aria-labelledby` or `aria-describedby` for the heading and description of the `AlertDialog` and maybe show some of those in the code highlights above?

Author: Thanks @aramos-adobe!

    </sp-button>
    <sp-button
        slot="button"
        variant="negative"

>         treatment="fill"

Reviewer: ```suggestion
        treatment="outline"
```

aligning to the design specs

</sp-alert-dialog>
```

</sp-tab-panel>
</sp-tabs>

> ### Behaviors

Reviewer: Lets add a behavior section that should the alert dialog in an overlay with the correct accessibility set up. we can make a follow up ticket to add it to storybook and clean up the button spacing issue

    </sp-button>
</sp-alert-dialog>
```

> ### Accessibility

Reviewer: Lets break this up in to sub sections? Maybe leverage the keyboard commands table? lets chat about a reusable keyboard table?

The `<sp-alert-dialog>` component follows these accessibility guidelines:

-   Uses semantic heading elements (`<h2>`) for the dialog title
-   Ensures all buttons have clear, descriptive labels
-   Supports keyboard navigation:
    -   <kbd>Tab</kbd>: Move focus between buttons
    -   <kbd>Space</kbd>/<kbd>Enter</kbd>: Activate the focused button
    -   <kbd>Esc</kbd>: Close the dialog
-   Maintains proper focus management when opened and closed

> -   Uses ARIA roles and attributes appropriately:

Reviewer: Should our examples have `role` defined based on this guidance?

Author: @caseyisonit IMHO this should be part of what the component does for the consumer, rather than the onus be on the consumer. Also, I'm interested in having a11y auditors confirm that modal alerts are `alertdialog` because they interrupt user action where `alert` roles do not.
